mongo/collection.go

```diff
@@ -1223,8 +1335,19 @@ func (coll *Collection) FindOneAndDelete(ctx context.Context, filter interface{}
 	return coll.findAndModify(ctx, op)
 }
 
-// FindOneAndReplace finds a single document and replaces it, returning either
-// the original or the replaced document.
+// FindOneAndReplace executes a findAndModify operation
+// (https://docs.mongodb.com/manual/reference/command/findAndModify/) to replace at most one document in the collection.
+//
+// The filter parameter should be a document containing query operators and can be used to select the document to be
+// replaced. It cannot be nil. An empty document (e.g. bson.D{}) will select one of the documents in the collection. If
+// the filter does not match any documents, a SingleResult with an error set to ErrNoDocuments wil be returned. If the
+// filter matches multiple documents, one will be selected from the matched set.
```

Comment:
Ditto on select and sort.

Reply:
resolving so we can continue conversation in the `Find` thread.